### ISSUE TITLE
Buildah cannot create WORKDIR directory

### ISSUE THREAD

**Issue by Aeron** [actions/runner-images#9427]
### Description

The issue is simple: during the image-building process (with the [redhat-actions/buildah-build@v2](https://github.com/redhat-actions/buildah-build/tree/v2/) action), a `WORKDIR` directive with a nonexistent directory will end up with an `mkdir` error.

It’s a known issue ([containers/buildah#2475](https://github.com/containers/buildah/issues/2475)), and it‘s been considered fixed ([containers/podman#7239](https://github.com/containers/podman/pull/7239)) for a long time. But the action [doesn’t use its own images](https://github.com/redhat-actions/buildah-build/issues/77#issuecomment-938042345) and relies on a runner.

The current version of Buildah on `ubuntu-latest` is `1.23.1`, but it’s unclear when the fix was released. The Buldah and Podman change logs and releases have nothing about the related issues or PRs.

So, it’d be better to update Buildah as far as possible.

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 11
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [ ] macOS 14
- [ ] macOS 14 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Runner image version is `20240225.1.0`.

Here is the link to a failed build’s exact error: https://github.com/Aeron/mimalloc-container/actions/runs/8084278971/job/22092474617#step:5:90.

### Is it regression?

I’m not sure. I’ve never encountered it before.

### Expected behavior

A build finishes without `mkdir` (or any) errors.

### Actual behavior

A build ends up with code 125 and `mkdir: not a directory` error.

### Repro steps

1. Install `buildah` version `1.23.1`
2. Create a Dockerfile with the `WORKDIR` directive pointing to a nonexistent directory, like so:
    ```Dockerfile
    FROM alpine:latest
    WORKDIR /usr/src/nonexistent
    ```
3. Build it, like so:
    ```sh
    /buildah bud -f ./Dockerfile --format docker -t test .
    ```

**Comment by shamil-mubarakshin** [comment]
Hey @Aeron,
I couldn't reproduce the issue https://github.com/shamil-mubarakshin/tests-repository/actions/runs/8095451348.
`buildah` and `podman` are pre-installed on runner's image from ubuntu repository using apt and at this time we have no plans of changing the installation method. Given that issue you are referring to was reported in 2020, I would expect it to be delivered to `buildah` (v.1.23.1 released on 09.2021) and `podman` (v3.4.4 released on 12.2021).
I will be closing the issue for now. Feel free to reach out in case of further concerns.

**Comment by Aeron** [comment]
@shamil-mubarakshin, I agree the timeline of issues, fixes, and releases looks strange, but I don’t have another explanation.

I’m not sure why the short variant of steps-to-reproduce didn’t yield the same results. I didn’t test the example but assumed it—that’s on me. 

And yet you have a link to my stuff, you can use it, it’s very straightforward, there’s no magic there. You don’t need to debug my baggage or anything like it. You can test the exact Dockerfile to be 100% sure. It should’ve been step two, not my assumption. (I’ll edit it.)

That’s the thing: Docker’s `build` has no problem with the original Dockerfile, even the clean run without a cache or anything. So, Buildah shouldn’t either, but it does.

To be 146% sure, I’ve run an Ubuntu 22.04 VM, installed Buildah (it’s the same `1.23.1 (image-spec 1.0.1, runtime-spec 1.0.2-dev)` and run the build command. And it ended up with the same error:

```text
[1/2] STEP 6/8: WORKDIR /usr/src/mimalloc/build
error building at STEP "WORKDIR /usr/src/mimalloc/build": copier: mkdir: error checking directory "/usr/src/mimalloc/build": mkdir /usr/src/mimalloc/build: not a directory
```
